File: JDK11/opentraceability/src/main/java/opentraceability/models/events/kdes/EventKDEDateTime.java
package opentraceability.models.events.kdes;

import opentraceability.interfaces.IEventKDE;
import opentraceability.utility.XElement;
import org.json.JSONObject;

import java.time.OffsetDateTime;
import org.w3c.dom.Element;

public class EventKDEDateTime extends IEventKDE {
    public Class valueType = OffsetDateTime.class;

    public OffsetDateTime value = null;

    public EventKDEDateTime() {
    }

    public EventKDEDateTime(String ns, String name) {
        this.namespace = ns;
        this.name = name;
    }

    @Override
    public Object getJson() {
        throw new UnsupportedOperationException("Not yet implemented");
    }

    @Override
    public XElement getXml() {
        throw new UnsupportedOperationException("Not yet implemented");
    }

    @Override
    public void setFromJson(Object json) {
        throw new UnsupportedOperationException("Not yet implemented");
    }

    @Override
    public void setFromXml(XElement xml) {
        throw new UnsupportedOperationException("Not yet implemented");
    }

    @Override
    public String toString() {
        if (value != null) {
            return value.toString();
        } else {
            return "";
        }
    }
}

File: Java/OpenTraceability/src/main/opentraceability/models/events/kdes/eventkdedatetime.kt
package models.events.kdes

import com.fasterxml.jackson.core.JsonToken
import interfaces.IEventKDE
import org.json.simple.JSONObject
import java.lang.reflect.Type
import java.time.OffsetDateTime
import org.w3c.dom.Element

class EventKDEDateTime: EventKDEBase, IEventKDE {

    override var ValueType: Type = OffsetDateTime::class.java

    var Value: OffsetDateTime? = null

    constructor(ns: String, name: String) {
        this.Namespace = ns;
        this.Name = name;
    }


    override fun GetJson(): JSONObject? {
        TODO("Not yet implemented")
    }

    override fun GetXml(): Element? {
        TODO("Not yet implemented")
    }

    override fun SetFromJson(json: JSONObject) {
        TODO("Not yet implemented")
    }

    override fun SetFromXml(xml: Element) {
        TODO("Not yet implemented")
    }

    override fun toString(): String {
        if (Value != null) {
            return Value.toString()
        } else {
            return ""
        }
    }
}
